src/AzSK.ADO/Framework/BugLog/BugLogCheckerHelper.ps1

```diff
@@ -0,0 +1,179 @@
+                        $azTableBugInfo[0].IsDeleted = "Y";
+                        $azTableBugInfo[0] | Update-AzTableRow -Table $cloudTable;
+                    }
+
+                    return $bugObj;
+                }
+            }
+        }
+        catch {
+            Write-Host "Not able to access storage account."
+        }
+        
+        return $bugObj;
+    }
+
+    hidden [bool] InsertBugInfoInTable([string] $hash, [string] $projectName, [string] $ADOBugId) 
+    {
+        try 
+        {
+           $tableName = $this.GetTableName($projectName);
+
+           #Get table filterd by name.
+           $storageTables = Get-AzStorageTable -Context $this.StorageAccountCtx | Select Name;
+
+           #create table if table not found.
+           if ( !$storageTables -or ($storageTables.Count -eq 0) -or !($storageTables.Name -eq $tableName) ) {
+               New-AzStorageTable -Name $tableName -Context $this.StorageAccountCtx;   
+           }
+           #Get cloudTable to do some operations on the table.
+           $cloudTable = (Get-AzStorageTable -Name $tableName -Context $this.StorageAccountCtx).CloudTable;
+
+           #Add data in table.
+           $partitionKey = $hash;
+           $rowKey = $hash + "_" + $ADOBugId;
+           Add-AzTableRow -Table $cloudTable -PartitionKey $partitionKey -RowKey ($rowKey) -property @{"ADOBugId" = $ADOBugId; "ADOScannerHashId" = $hash; "IsDeleted" = "N"; "ProjectName" = $projectName};
+           return $true;           
+        }
+        catch {
+            return $false;
+        } 
+    }
+
+    hidden [bool] GetTableEntityAndCloseBug([string] $hash) 
+    {    
+        #get table filter by name
+        $tableName = $this.GetTableName("");
+
+        try {
+            #get storage table.
+            $cloudTable = (Get-AzStorageTable -Name $tableName -Context $this.StorageAccountCtx).CloudTable;
+            if ($cloudTable) {
+                #Get clouddata to do some operation on the table
+    
+                #$azTableBugInfo += Get-AzTableRow -table $cloudTable -columnName "ADOScannerHashId" -value $hash -operator Equal 
+                $azTableBugInfo = @();
+                $azTableBugInfo += Get-AzTableRow -table $cloudTable -CustomFilter "(($hash) and (IsDeleted eq 'N'))";
+
+                if ($azTableBugInfo -and $azTableBugInfo.count -gt 0) {
+                    foreach ($row in $azTableBugInfo) {
+                        if($this.CloseBug($row[0].ADOBugId, $row[0].projectName) )
+                        {
+                            $row.IsDeleted = "Y";
+                            $row | Update-AzTableRow -Table $cloudTable;
+                        }
+                    } 
+                }
+            }
+        }
+        catch {
+            if (!$this.errorMsgDisplayed) {
+               Write-Host "Not able to access storage account. Could not close the bug." -Red  
```

Comment:
Could not update entry of closed bug in storage table.

Reply:
changed